### PR TITLE
UCT/API: Add configuration to CM

### PR DESCRIPTION
**Pull request by alinask**
**What**
Add the option to configure different values for a CM (connection manager) .

**Why**
To allow the user to set and change the CM's parameters.

**How**
Add configuration options to the uct_cm_open api call and add cm config tables to the uct components.

### CODE REVIEW

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>FAILED</b> on 1 of 21 workers (click for details)</summary>

**Note:** the logs will be deleted after 03-Dec-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/a360e7cd6af7682ec0d77ee948a459dc/raw/e9f9cbdc4818285c42d2ade613713b3a52d9272a/_main) | :question: ABORTED
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/a360e7cd6af7682ec0d77ee948a459dc/raw/7eb45f6f7d6713d4740d6932b325e8e80ec85190/hpc-arm-hwi-jenkins_W0) | :question: ABORTED
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/a360e7cd6af7682ec0d77ee948a459dc/raw/032612486ce99c94160c5da083942cca9a664f05/hpc-arm-hwi-jenkins_W1) | :question: ABORTED
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/a360e7cd6af7682ec0d77ee948a459dc/raw/10ae6b04d413016560208a9d190287482faad21c/hpc-arm-hwi-jenkins_W3) | :question: ABORTED
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/a360e7cd6af7682ec0d77ee948a459dc/raw/86c2a3beca819a75ccd31fcd5c14a59ff17fc05b/hpc-arm-hwi-jenkins_W2) | :x: FAILURE
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/a360e7cd6af7682ec0d77ee948a459dc/raw/5f1ebff4b310268b2d496d9d18a8455044b85a5f/hpc-arm-cavium-jenkins_W0) | :question: UNKNOWN
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/a360e7cd6af7682ec0d77ee948a459dc/raw/2d1d63c139f51e909ce2aae03c15f0eab2fd5b95/hpc-arm-cavium-jenkins_W1) | :question: UNKNOWN
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/a360e7cd6af7682ec0d77ee948a459dc/raw/5600e1e2bc6c9b4c8af42c6d72d18ea1e06e20a3/hpc-arm-cavium-jenkins_W2) | :question: UNKNOWN
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/a360e7cd6af7682ec0d77ee948a459dc/raw/421faaeb56d574b74612b68ef80c9d5e26dc3f7e/hpc-arm-cavium-jenkins_W3) | :question: UNKNOWN
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/a360e7cd6af7682ec0d77ee948a459dc/raw/b02be974aed481a1c2f32451c8a77114bc21c728/hpc-test-node-gpu_W0) | :question: UNKNOWN
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/a360e7cd6af7682ec0d77ee948a459dc/raw/c351a3e5ca9d93c2eb020a0a882602b2b95b9eb0/hpc-test-node-gpu_W1) | :question: UNKNOWN
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/a360e7cd6af7682ec0d77ee948a459dc/raw/faad3773e099c70deec10ef02e207e136dceda01/hpc-test-node-gpu_W2) | :question: UNKNOWN
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/a360e7cd6af7682ec0d77ee948a459dc/raw/e32895782ad3ec936eae97f6950c3bc29bbbe91a/hpc-test-node-gpu_W3) | :question: UNKNOWN
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/a360e7cd6af7682ec0d77ee948a459dc/raw/6a5f531c9b05bece288cf87fb7b19819d09fb259/hpc-test-node-legacy_W3) | :question: UNKNOWN
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/a360e7cd6af7682ec0d77ee948a459dc/raw/623d9440474eb9d62dab8cc95f094f8495fbe856/hpc-test-node-new_W0) | :question: UNKNOWN
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/a360e7cd6af7682ec0d77ee948a459dc/raw/d45380e1357ff4ec0991aefd5c2bb0c1af7894be/hpc-test-node-new_W1) | :question: UNKNOWN
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/a360e7cd6af7682ec0d77ee948a459dc/raw/f7aa56c0f060b6b9f115dd08b8f53e3b59530d5a/hpc-test-node-new_W2) | :question: UNKNOWN
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/a360e7cd6af7682ec0d77ee948a459dc/raw/44809a5507ebfed0550a1352d38d246a549ba6a7/hpc-test-node-new_W3) | :question: UNKNOWN
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/a360e7cd6af7682ec0d77ee948a459dc/raw/7d6fbfc639c9bec8a38b1efa51b3fbcdb4e15dbb/r-vmb-ppc-jenkins_W1) | :question: UNKNOWN
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/a360e7cd6af7682ec0d77ee948a459dc/raw/e3778d17cf959d6fcbfc85765f2c911dafb9de80/r-vmb-ppc-jenkins_W2) | :question: UNKNOWN
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/a360e7cd6af7682ec0d77ee948a459dc/raw/cf31f57f643b0c96c3dddb7b9a1925c779011d0d/r-vmb-ppc-jenkins_W3) | :question: UNKNOWN

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>PASSED</b#  > on 25 workers (click for details)</summary>

**Note:** the logs will be deleted after 03-Dec-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/aa33366ecb7d601537f44ce8c1f681d8/raw/11df9bc884f8d0516cd96503a70f72fd79af3b93/_main) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/aa33366ecb7d601537f44ce8c1f681d8/raw/4ea0f7e4407b77f709efdace92212378d338369d/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/aa33366ecb7d601537f44ce8c1f681d8/raw/00b6be63dafaf0423528c801eb95f33bb26c587f/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/aa33366ecb7d601537f44ce8c1f681d8/raw/9a008b2997f5154b33f4889a01e25801cb2e22a7/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/aa33366ecb7d601537f44ce8c1f681d8/raw/cc41c50d174e8935e14a850b6731c4d6ad326f41/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/aa33366ecb7d601537f44ce8c1f681d8/raw/555485f49def9ef73ae1ca68d1f762d248aeae73/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/aa33366ecb7d601537f44ce8c1f681d8/raw/1a413d46d571515ea80bde3cbdba920020d0d7b8/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/aa33366ecb7d601537f44ce8c1f681d8/raw/938df6c04f953b0e01cd01f808eb66115db48964/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/aa33366ecb7d601537f44ce8c1f681d8/raw/2659a124c2fdab99aa40be6f596c8e5f5e10da7c/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/aa33366ecb7d601537f44ce8c1f681d8/raw/e58fef35dc81666b8b724030db56509b46c1f072/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/aa33366ecb7d601537f44ce8c1f681d8/raw/ba91d060fed623d8e73657cbea809f9d6a20b785/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/aa33366ecb7d601537f44ce8c1f681d8/raw/11bd5bc5cd0d1e704064f416fc708a339edc6e01/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/aa33366ecb7d601537f44ce8c1f681d8/raw/c38a17607d910224ef04f2fa19b585a061e52598/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/aa33366ecb7d601537f44ce8c1f681d8/raw/d2721c047d3d98878b5b3034cdcf2e4cd7161181/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/aa33366ecb7d601537f44ce8c1f681d8/raw/cb8e2f0a209ffb0f2b7045c5fdb2c42cb6c7a1c3/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/aa33366ecb7d601537f44ce8c1f681d8/raw/a3acb920ecb97aa496095af03035c2ea32780d01/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/aa33366ecb7d601537f44ce8c1f681d8/raw/c266262d3aa5387798c1b446b49f370c971c7b1b/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/aa33366ecb7d601537f44ce8c1f681d8/raw/1c871c8a2fae587bf50e5811db8b497977ef3256/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/aa33366ecb7d601537f44ce8c1f681d8/raw/3382bc16821ec5c0cb0bf1d79b0668d430733255/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/aa33366ecb7d601537f44ce8c1f681d8/raw/7346125f626e7a2c0c1c53dcc1fa3b0878477082/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/aa33366ecb7d601537f44ce8c1f681d8/raw/2adc398ee817b808bb2025a407c407731398b5ee/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/aa33366ecb7d601537f44ce8c1f681d8/raw/fa95b58e9afe2389687608b33d184560948e8caa/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/aa33366ecb7d601537f44ce8c1f681d8/raw/53c289b8808b75a7b85bd82182848c0a3847b6b5/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/aa33366ecb7d601537f44ce8c1f681d8/raw/3be7d817c50c061c0132437e57b37bc631e9e1dc/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/aa33366ecb7d601537f44ce8c1f681d8/raw/7112c83a5fd55160323483d9be04650d9689480b/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

**Comment by yosefe** [comment]
bot:pipe:retest

File: src/uct/base/uct_md.h [hunk 1]
```diff
@@ -20,6 +20,17 @@ typedef struct uct_md_rcache_config {
     double               overhead;     /**< Lookup overhead estimation */
 } uct_md_rcache_config_t;
 
+/**
+ * Keeps information about allocated configuration structure, to be used when
+ * releasing the options.
+ */
+typedef struct uct_config_bundle {
```
Comment (yosefe) [inline src/uct/base/uct_md.h]: move this to component.h

File: src/uct/base/uct_md.h [hunk 2]
```diff
@@ -118,6 +129,10 @@ uct_md_fill_md_name(uct_md_h md, void *buffer)
 #endif
 }
 
+ucs_status_t uct_config_read(uct_config_bundle_t **bundle,
```
Comment (yosefe) [inline src/uct/base/uct_md.h]: move this to component.h

File: test/gtest/uct/uct_test.cc [hunk 3]
```diff
@@ -660,17 +664,34 @@ uct_test::entity::entity(const resource& resource, uct_md_config_t *md_config) {
...
+    comp_attr.field_mask = UCT_COMPONENT_ATTR_FIELD_NAME |
+                           UCT_COMPONENT_ATTR_FIELD_FLAGS;
+    status = uct_component_query(resource.component, &comp_attr);
     ASSERT_UCS_OK(status);
 
-    max_conn_priv = 0;
+    if (comp_attr.flags & UCT_COMPONENT_FLAG_CM) {
+        status = uct_cm_config_read(resource.component, NULL, NULL, &cm_config);
+        ASSERT_UCS_OK(status);
+
```
Comment (yosefe) [inline test/gtest/uct/uct_test.cc]: redundant space

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>FAILED</b> on 3 of 25 workers (click for details)</summary>

**Note:** the logs will be deleted after 05-Dec-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/7107fd188f5b91a64672df79ded8dfaa/raw/70e0e3d4572056c8a925c6e3e283d63dbddc9d15/_main) | :x: FAILURE
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/7107fd188f5b91a64672df79ded8dfaa/raw/a6ad622e7379d6999d662f89300ceba90060c837/hpc-test-node-legacy_W0) | :x: FAILURE
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/7107fd188f5b91a64672df79ded8dfaa/raw/65e334d66030ba30be992ceb06709e849f2f4ce1/hpc-test-node-legacy_W3) | :x: FAILURE
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/7107fd188f5b91a64672df79ded8dfaa/raw/3350e4880f3bf1b0707d4cec56f0cc5917030825/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/7107fd188f5b91a64672df79ded8dfaa/raw/29f73946e42e064ec4bd005eae7bf31d3bdc6bb6/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/7107fd188f5b91a64672df79ded8dfaa/raw/3d7db141020c6f2246d11863acce261d666bb4c4/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/7107fd188f5b91a64672df79ded8dfaa/raw/bb23ccf7dddd0c93a847a85a0958f43c53a5514f/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/7107fd188f5b91a64672df79ded8dfaa/raw/b47b73ab5d597a1cbc5856b480b3f573c6fee800/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/7107fd188f5b91a64672df79ded8dfaa/raw/368d9a87c7d9cb8469eac6a01dc75196aebe54bc/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/7107fd188f5b91a64672df79ded8dfaa/raw/c42a4630d69eb6d696aad1d6a4c594f22dcfedb7/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/7107fd188f5b91a64672df79ded8dfaa/raw/3e574f06c569ef05e80041252c4534cc5735d08f/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/7107fd188f5b91a64672df79ded8dfaa/raw/30de52e8a1a909eadad1342ba109b00fd3ef3ba4/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/7107fd188f5b91a64672df79ded8dfaa/raw/382e0fe4014b4424de0cd55c009fc032c9ef8ca5/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/7107fd188f5b91a64672df79ded8dfaa/raw/cbd13bd48e38ea9ad4ae91ec2c118d386510d96e/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/7107fd188f5b91a64672df79ded8dfaa/raw/dca5490f0f7aed09dc8a0a940f29514725497e81/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/7107fd188f5b91a64672df79ded8dfaa/raw/0833fa57a1397894c865239c8c24422cce828eb0/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/7107fd188f5b91a64672df79ded8dfaa/raw/5653765b94cf6c1ea957e912e86b67320a55cfa2/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/7107fd188f5b91a64672df79ded8dfaa/raw/8c1845ce229b8b259c01107e812fbe65b01f1538/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/7107fd188f5b91a64672df79ded8dfaa/raw/42df72609dab0585e60652105bee7e15df875217/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/7107fd188f5b91a64672df79ded8dfaa/raw/01ded9f3800f5a05d759c1cede6fdd9a63000cf1/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/7107fd188f5b91a64672df79ded8dfaa/raw/928f5a751532850af245d2a0b447059b3ff77847/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/7107fd188f5b91a64672df79ded8dfaa/raw/9e07ce062da093599069fe5cb1fc2cac43260b78/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/7107fd188f5b91a64672df79ded8dfaa/raw/ac16e69a5fa672dbfd82ee48bc82e86f6b4be6e1/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/7107fd188f5b91a64672df79ded8dfaa/raw/0532a94784b4b009d071fcef1de4e3f9b4b6980c/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/7107fd188f5b91a64672df79ded8dfaa/raw/715c4137fd32672fb80296628d1045ad85cd7645/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

**Comment by alinask** [comment]
bot:pipe:retest

**Comment by alinask** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>PASSED</b#  > on 25 workers (click for details)</summary>

**Note:** the logs will be deleted after 05-Dec-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/bcfd251df1dc45036898ddf8f832a5da/raw/d2f1a44752bc72d37903ab37beaf382242450d13/_main) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/bcfd251df1dc45036898ddf8f832a5da/raw/9c57248c799c958e018681a7bb06aa3dbe576de8/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/bcfd251df1dc45036898ddf8f832a5da/raw/1e1bc7cf2c05c02d70704e0efa303296e852e84d/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/bcfd251df1dc45036898ddf8f832a5da/raw/e05beeadf9572af4c90d92e8769907062abe6a11/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/bcfd251df1dc45036898ddf8f832a5da/raw/1fc8bc102c7d8aa796a13a5fb72fb548e193117f/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/bcfd251df1dc45036898ddf8f832a5da/raw/955e5946ac1c869b7642f22a34e5eb26b599f606/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/bcfd251df1dc45036898ddf8f832a5da/raw/e86fd8fc95b738f933b2efa9b05cff287df50d43/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/bcfd251df1dc45036898ddf8f832a5da/raw/2760b0dd1c43154beafbde40fd7e1bcdb33a083e/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/bcfd251df1dc45036898ddf8f832a5da/raw/25c8106b876500a31fe48d8cbbc0e032670ed43d/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/bcfd251df1dc45036898ddf8f832a5da/raw/f64b702656ca72079425fe4645d9694284ab74da/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/bcfd251df1dc45036898ddf8f832a5da/raw/8bc4fe7125582749e0faf6863e93cd29a2f90b8a/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/bcfd251df1dc45036898ddf8f832a5da/raw/da531847481e9dfead4eebb81cf98e99a97cf07a/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/bcfd251df1dc45036898ddf8f832a5da/raw/f2b8c057d48fcdefdfdcb22f259c7c324463ec6c/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/bcfd251df1dc45036898ddf8f832a5da/raw/2e79677a973c17eb832731d9e3134883d272e465/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/bcfd251df1dc45036898ddf8f832a5da/raw/c3328e3ff3ca329f039bdef2613d6c4104bff7da/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/bcfd251df1dc45036898ddf8f832a5da/raw/f692406b47f40966da3821a67cf08627a44b2b10/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/bcfd251df1dc45036898ddf8f832a5da/raw/1a8cb5cf5e2facae17b1fae366e2638f91db4833/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/bcfd251df1dc45036898ddf8f832a5da/raw/1516fee82ae6da9d0eb4b3aae70acd544e94d143/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/bcfd251df1dc45036898ddf8f832a5da/raw/e0ae7d4f080417ecb906d263de7b7152c9000b16/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/bcfd251df1dc45036898ddf8f832a5da/raw/c8554913753ca54bd610306afe67fd53ae98ac62/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/bcfd251df1dc45036898ddf8f832a5da/raw/81e6245b7cc7798306b69ad9975a5bb66ba4f729/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/bcfd251df1dc45036898ddf8f832a5da/raw/d38c38de1a0e011fed3f3911f273cd5b9a45f5d7/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/bcfd251df1dc45036898ddf8f832a5da/raw/b9575796787751415c9850ad275051f7990e2816/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/bcfd251df1dc45036898ddf8f832a5da/raw/c116f1dd64a969f57149f5810986f462ea2740c3/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/bcfd251df1dc45036898ddf8f832a5da/raw/bf0ef02ba1044fa2262e5044ae6944f223158400/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

**Comment by yosefe** [comment]
@shamisp @bbenton can you pls take a look?

**Comment by yosefe** [comment]
@shamisp @bbenton ping

**Comment by shamisp** [comment]
@yosefe why/what/how is missing.

**Comment by shamisp** [comment]
Do you know if this going to break Open MPI again ?

**Comment by shamisp** [comment]
Does it make sense to keep the version with original function signature that will call the new one with NULL config ?

**Comment by yosefe** [comment]
@shamisp btl/uct is not using CM, it's new API introduced in v1.7

**Comment by alinask** [comment]
@shamisp why/what/how was added.

**Comment by alinask** [comment]
@shamisp is this PR good to go?

**Comment by alinask** [comment]
@shamisp ping

**Comment by shamisp** [comment]
@yosefe I guess it was by mistake - let's revert and finish standard procedure

File: src/uct/api/uct.h [hunk 4]
```diff
@@ -3000,11 +3000,15 @@ UCT_INLINE_API unsigned uct_iface_progress(uct_iface_h iface)
  * @param [in]  component   Component on which to open the connection manager,
  *                          as returned from @ref uct_query_components.
  * @param [in]  worker      Worker on which to open the connection manager.
+ * @param [in]  config      CM configuration options. Should be obtained
+ *                          from uct_cm_config_read() function, or point to
+ *                          CM-specific structure which extends uct_cm_config_t.
```
Comment (bbenton) [inline src/uct/api/uct.h]: Plz add ref for uct_cm_config_t

Comment (alinask) [reply src/uct/api/uct.h]: @bbenton Added - https://github.com/openucx/ucx/pull/4574

**Comment by shamisp** [comment]
We also have to run this by @tonycurtis. Also, do we we have field bit mask for config. My preference to roll it back instead of patching over with #4574.

**Comment by tonycurtis** [comment]
src/uct/api/uct.h

@param [in] filename If non-NULL, read configuration from this file. If the file does not exist, it will be ignored.

Add text to include if file exists, but can’t be opened/read?

@param [in] config CM configuration options. Should be obtained      
from uct_cm_config_read() function, or...

I’d change “should be” to “either” or “can be”.  And “which” -> “that” later in sentence.

Tony

**Comment by alinask** [comment]
> We also have to run this by @tonycurtis. Also, do we we have field bit mask for config. My preference to roll it back instead of patching over with #4574.

@shamis we don't use field bit mask for config tables.

**Comment by alinask** [comment]
> src/uct/api/uct.h @param [in] filename If non-NULL, read configuration from this file. If the file does not exist, it will be ignored. Add text to include if file exists, but can’t be opened/read? @param [in] config CM configuration options. Should be obtained from uct_cm_config_read() function, or... I’d change “should be” to “either” or “can be”. And “which” -> “that” later in sentence. Tony

@tonycurtis Your comments are applied  - https://github.com/openucx/ucx/pull/4574

**Comment by shamisp** [comment]
@alinask - why ? Do you have any example of configuration option ? I just see place holder and no config options ?

**Comment by alinask** [comment]
@shamisp This is the same config usage we also have in uct_md_open and uct_iface_open for example. 'struct uct_cm_config' is a base structure that other CMs should extend for their own config.
This PR doesn't have this last part because this is mainly an API PR, I added it in this PR - https://github.com/openucx/ucx/pull/4573 , here specifically - https://github.com/openucx/ucx/pull/4573/files#diff-f0a5f4b1069c01b0a0178ce8f0c64ec9R22

**Comment by shamisp** [comment]
BTW, this will break open MPI again. Is this change absolutely necessary ?  I would suggest introduce API/ABI versioning and only then introduce this API change. Also, at this point I suggest to maintain original API version as well that will use some difficult configuration.

**Comment by alinask** [comment]
@shamisp Why would this break OMPI? OMPI doesn't use the CM API.

**Comment by shamisp** [comment]
@alinask I'm discussing with Nathan. The patch didn't follow community procedure and will be rollbacked. On re-submission, please combine the PR with #4574. Also, as a use case you didn't specify what parameters exactly you want to configure ? Typically it will be good to have an actual implementation but at least justification of the use case has to be provided.

**Pull request merged**